### ISSUE TITLE
Add support for using type alias.

### ISSUE THREAD

**Issue by IOBYTE** [paul-j-lucas/cdecl#1]
Are there any plans to add C++ using type aliases?

```
c++decl> explain typedef int v3[3];
declare v3 as type array 3 of int
c++decl> explain using v3 = int[3];
                 ^
9: syntax error: "using": unexpected token
c++decl>
```

**Comment by paul-j-lucas** [comment]
For some reason, I don't get notified of new issues when they're created, so I didn't see this until just now just _after_ I published **cdecl** 4.5 that contains support for C++11's `using` declarations. However, I forgot to add support for _explaining_ `using` declarations. I'll get around to it.

I also discovered that there are some broken `using` cases like the one you gave. :(

**Comment by paul-j-lucas** [comment]
OK, I just added support for `explain using` _name_ `=` _gibberish_.

I also realized that the breakage isn't limited to `using`. It fails to parse array declarations without a name, e.g., `explain void f(char[])`. If you insert a name like `explain void f(char s[])` then it works.

Unfortunately, it's not an easy fix and will likely require some serious reworking of parts of the grammar so as not to trigger a gazillion more shift/reduce conflicts (as well as new reduce/reduce conflicts).

With 671 unit tests, you'd think I would have caught this earlier; but no. :(

**Comment by paul-j-lucas** [comment]
> Unfortunately, it's not an easy fix and will likely require some serious reworking of parts of the grammar so as not to trigger a gazillion more shift/reduce conflicts (as well as new reduce/reduce conflicts).

It turns out that the fix was simpler than I thought. It's fixed in 4.6.1.
